### PR TITLE
[ECAM] Change slats, wing and flaps shapes

### LINKED ISSUE #1803: [REQUEST] ECAM Slat symbol redesign
**Is your feature request related to a problem? Please describe.**
No.

**Describe the feature you'd like to see implemented**

As @stevex pointed out during #1549 analisys, the left part of slats symbol should look slightly different from the flaps symbol - the most left edge shouldn't be strictly vertical, rather the symbol should be more of "diamantish" shape.

**References**

A32X ECAM:
- Slats retracted:
image
- Slats extended:
image

IRL images:
- Slats retracted:
image
- Slats extended:
image

**Additional context**

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by paul92ilm**
Fixes #1803 

## Summary of Changes

- Improve slats design --> diamond shape
- Improve wing/grey shape --> non symmetric shape
- Improve flaps shape --> now closed shape (first point = last point) 
- Improve design and positions for both slats and flaps dots
- Improve lever position and flaps/slats indicators when they are in transit

#### Before
image
image

#### After
image
image

## References

image
image

## Additional context

Discord username (if different from GitHub): Lussion

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by aviationenthusiast** [comment]
> image

does the grey quadrilateral in the middle disappear like the second image in references?

**Comment by paul92ilm** [comment]
Not yet, I'm working on the slats and flaps dots before.
Will work on the wing/grey shape after

**Comment by paul92ilm** [comment]
> > > image
> > 
> > 
> > does the grey quadrilateral in the middle disappear like the second image in references?
> 
> It does not disappear. It looks like so but it was just the camera inability to show all colours correctly.
> Original reference
> https://youtu.be/aAvQXMbiodw
> Look at 00:20

👍 
So it's ready for reviews and tests

**Comment by stevex77** [comment]
Tested with following results:

FLAPS RETRACTED:
F0

FLAPS 0 >>> 1+F:
F0+

FLAPS 1+F:
F1

FLAPS 1+F >>> 2:
F1+

FLAPS 2:
F2

FLAPS 2 >>> 3:
F2+

FLAPS 3:
F3

FLAPS 3 >>> FULL:
F3+

FLAPS FULL:
FF

FLAPS FULL >>> 3:
FF-

FLAPS 3 >>> 2:
F3-

FLAPS 1+F >>> RETRACTED:
F1-

---

Reference (retracted):
FREF

Result: Great work!

**Comment by stevex77** [comment]
I believe that this gray quadrilateral should remain there.
I adjusted the exposure of reference video and it's clearly visible:

FLAPS

So from my POV this is implemented correctly. :)

**Comment by lukecologne** [comment]
According to the FCOM, the selected (but not reached) slat/flap position should have a blue diamond below the normal white diamond, instead of the current blue flaps indicator. Do you want to implement that as well, and if yes, also in this PR? It's OK if you don't want to. References: 
grafik
grafik
https://youtu.be/g32B8_5TbdQ?t=80

**Comment by paul92ilm** [comment]
Will work on it

**Comment by stevex77** [comment]
Have found this video, but it's not NEO excactly. But should be the same for NEO when comparing video to FCOM snip @lukecologne attached.

I also noticed that "lever position" indication (5) should possibly be CYAN when flas / slats in transit based on bot the video and FCOM snip.

https://www.youtube.com/watch?v=TgETvlt3QjY

**Comment by Benjozork** [comment]
@stevex77, it seems like you don't have the QA template - can you give me your discord username so I can have the QA role given to you ? you seem to help out a lot

**Comment by stevex77** [comment]
@Benjozork sure, my discord is SteveX77 [Z+0!]#1407
Already assigned to QA TESTERS group there.
And yes, QA template will be usefull. :)

**Comment by paul92ilm** [comment]
Cyan indicator replaced
image

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 1]
```diff
@@ -59,10 +59,14 @@ var A320_Neo_UpperECAM;
...
-        const _h = 6;
-        const _ox = 2;
-        const _oy = 1;
+        const _xOffset = 3.2;
+        const _yOffset = 1;
+        const _w = 3;
+        const _h = 4;
+        const _ox = 1.5;
+        const _oy = 1.5;
+        _cx = _cx + _xOffset;
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:68]: It's not a good idea to re-assign function parameters. You should create a new const instead.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js [hunk 2]
```diff
@@ -71,11 +75,15 @@ var A320_Neo_UpperECAM;
...
-        const _h = 5;
+        const _xOffset = -1;
+        const _yOffset = -0.5;
+        const _w = 3.5;
+        const _h = 3.5;
         const _ox = 2;
-        const _oy = -3;
-        const _ppoints = String(_cx - _w) + "," + String(_cy - _oy) + " " + String(_cx - _w - _ox) + "," + String(_cy - _h) + " " + String(_cx + _w) + "," + String(_cy + _oy) + " " + String(_cx + _w + _ox) + "," + String(_cy + _h);
+        const _oy = -2;
+        _cx = _cx + _xOffset;
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js:84]: Same as above

**Review by lukecologne — CHANGES_REQUESTED** [review]
Looks very nice, just a few small issues

**Comment by paul92ilm** [comment]
Final result
image

**Review by lukecologne — APPROVED** [review]
Alright, tested and looks good. If you don't want to add anything else, then it is good to go.

**Comment by paul92ilm** [comment]
> Alright, tested and looks good. If you don't want to add anything else, then it is good to go.

Points (1), (2) and (5) of your reference seems fully implemented. I will work on the 2 others for degraded configuration in an other PR.
So looks good to me to merge at this step

**Comment by lukecologne** [comment]
I don't think you need to worry about the rest, since they seem to rely on hydraulics or faults, both of which currently isn't implemented. You could take a look again once we start modeling these

**Pull request merged**